File: src/test/java/net/tassia/event/EventTest.java
package net.tassia.event;

import org.junit.Test;

public class EventTest {

	private boolean test = false, test2 = false;

	@Test
	public void test() {
		EventManager manager = new EventManager();
		manager.registerEvent(TestEvent.class);

		manager.registerListener(TestEvent.class, new TestListener());
		manager.registerListener(TestEvent.class, this::testListener);

		manager.callEvent(new TestEvent());

		assert test;
		assert test2;
	}

	private void testListener(TestEvent event) {
		test2 = true;
	}



	private static class TestEvent extends Event {
	}

	private class TestListener implements EventListener<TestEvent> {
		@Override
		public void onEvent(TestEvent event) {
			test = true;
		}
	}

}

File: src/test/kotlin/net/tassia/event/EventTest.kt
package net.tassia.event

import kotlin.test.Test

class EventTest {

	private class TestEvent : Event()

	private var test1 = false
	private var test2 = false

	@Test
	fun test() {
		val manager = EventManager.newDefault()
		manager.registerEvent(TestEvent::class)

		manager.registerListener(TestEvent::class, TestListener())
		manager.registerListener(TestEvent::class, this::testListener)

		manager.callEvent(TestEvent())

		assert(test1)
		assert(test2)
	}



	private fun testListener(event: TestEvent) {
		test2 = true
	}

	private inner class TestListener : EventListener<TestEvent> {
		override fun onEvent(event: TestEvent) {
			test1 = true
		}
	}

}
